_posts/en/newsletters/2025-02-07-newsletter.md

```diff
@@ -0,0 +1,284 @@
+  [anchor outputs][topic anchor outputs].  The script used determines
+  who can [CPFP fee bump][topic cpfp] the commitment transaction (and
+  under what conditions they can bump).  He presented four options:
+
+  - _Use a pay-to-anchor (P2A) script:_ this has a minimal onchain size
+    but means that all [trimmed HTLC][topic trimmed htlc] value will
+    probably go to miners (like it presently does).
+
+  - _Use a single-participant keyed anchor:_ this may allow some excess
+    trimmed HTLC value to be claimed by a participant who voluntarily
+    accepts waiting a few dozens blocks before being able to spend the
+    money they close out of a channel.  Anyone who wants to force close
+    a channel would need to wait that time anyway.  However, neither
+    party to the channel can delegate paying the fee to a third-party
+    without allowing that party to steal all of the channel funds.
+    If both you and your counterparty compete to claim the excess value,
+    it's likely that it will all go to miners anyway.
+
+  - _Use a shared key anchor:_ this also allows recovery of excess
+    trimmed HTLC value and allows delegation, but anyone who receives
+    the delegation can compete with you and your counterparty to claim
+    the excess value.  Again, in the case of competition, all excess
+    value is likely to go to miners.
+
+  - _Use a dual-keyed anchor:_ allows each participant to claim excess
+    trimmed HTLC value without having to wait additional blocks before
+    being able to spend.  However, it does not allow delegation.  The
+    two parties to a channel can still compete with each other.
+
+  In replies to the post, Gregory Sanders [noted][sanders ephanc] that
+  the different schemes could be used at different times.  For example,
+  P2A could be used when there were no trimmed HTLCs and one of the
+  keyed anchors could be used other times.  He also suggested that
+  trimmed value above the [dust threshold][topic uneconomical outputs]
+  could instead be added to the LN commitment transaction.
+  Additionally, we warned that it could create "new weirdness [a]
+  counterparty may be tempted to ramp up the trimmed amount and take it
+  themselves."  David Harding extended on this theme in a [later
+  post][harding ephanc].
+
+  Antoine Riard [warned][riard ephanc] against using P2A due to the risk
+  of encouraging miner [transaction pinning][topic transaction pinning]
+  (see [Newsletter #339][news339 pincycle]).
+
+  Discussion was ongoing at the time of writing.
+
+- **Emulating OP_RAND:** Oleksandr Kurbatov [posted][kurbatov rand] to
+  Delving Bitcoin about an interactive protocol that allows two parties
+  to make a contract that will pay out in a way neither of them can
+  predict, in other words: "randomly".  [Previous work][dryja pp] on
+  _probabilistic payments_ in Bitcoin has used advanced scripts, but
+  Kurbatov's approach entirely uses specially constructed public keys
```

Comment:
```suggestion
  Kurbatov's approach uses specially-constructed public keys
```
(or "uses only specially-constructed")

Reply:
Removed "entirely" but didn't add the dash.  My understanding is that words ending in -ly don't need a dash following them when used as an adjectival phrase.